File: frontend/demo/component/avatar/avatar-group-bg-color.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/avatar-group';
import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import { getPeople } from 'Frontend/demo/domain/DataService';
import { applyTheme } from 'Frontend/demo/theme';
import type Person from 'Frontend/generated/com/vaadin/demo/domain/Person';

@customElement('avatar-group-bg-color')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  @state()
  private items: Person[] = [];

  protected override async firstUpdated() {
    const { people } = await getPeople({ count: 6 });
    this.items = people;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-avatar-group
        .items="${this.items.map((person, index) => ({
          name: `${person.firstName} ${person.lastName}`,
          colorIndex: index,
        }))}"
      ></vaadin-avatar-group>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/avatar/AvatarGroupBgColor.java
package com.vaadin.demo.component.avatar;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.demo.domain.DataService;
import com.vaadin.demo.domain.Person;
import com.vaadin.flow.component.avatar.AvatarGroup;
import com.vaadin.flow.component.avatar.AvatarGroup.AvatarGroupItem;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import java.util.List;

@Route("avatar-group-bg-color")
public class AvatarGroupBgColor extends Div {

  private List<Person> people = DataService.getPeople(6);

  public AvatarGroupBgColor() {
    int colorIndex = 0;
    
    // tag::snippet[]
    AvatarGroup avatarGroup = new AvatarGroup();

    for (Person person : people) {
      String name = person.getFirstName() + " " + person.getLastName();
      AvatarGroupItem avatar = new AvatarGroupItem(name);
      avatar.setColorIndex(colorIndex++);
      avatarGroup.add(avatar);
    }
    // end::snippet[]

    add(avatarGroup);
  }

  public static class Exporter extends DemoExporter<AvatarGroupBgColor> { // hidden-source-line
  } // hidden-source-line
}
